### PR TITLE
Add old guild attribute to guild leave event

### PR DESCRIPTION
**Pull request by FasterSpeeding**
### Summary
Add old guild attribute to guild leave event

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

**Review by davfsa — APPROVED** [review]
LGTM

For reference, this is a little part of #766

**Comment by FasterSpeeding** [comment]
Since there's no "changes" section in the changelog i can't actually document changing GuildAvailableEvent.old_guild from Optional[Guild] to Optional[GatewayGuild]

**Comment by davfsa** [comment]
> Since there's no "changes" section in the changelog i can't actually document changing GuildAvailableEvent.old_guild from Optional[Guild] to Optional[GatewayGuild]

This would go under bugfix tbh

**Comment by FasterSpeeding** [comment]
> This would go under bugfix tbh

What was broken?

File: changes/806.feature.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Add `old_guild` attribute to `GuildLeaveEvent`.
+
```
Comment (davfsa) [inline changes/806.feature.md:2]: Trailing newline

**Review by davfsa — APPROVED** [review]

**Pull request merged**